### PR TITLE
display instagram pictures

### PR DESCRIPTION
**Pull request by lukas-menzel**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Cleanup (changes in structure but not functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested of the functionality with Node.js

**Test Configuration**:
* Go version:
* Kafka version:
* Kubernetes version:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: frontend/src/components/IGPost.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React, { Component, useState } from "react";
+import { withRouter, history } from "react-router";
+
+function IGPost({ post }) {
```
Comment (Urhengulas) [inline frontend/src/components/IGPost.js:4]: What are the `{`-brackets around the `post` for?

File: frontend/src/components/Result.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            </div>
+          </div>
+          <div className="column-one-sixed"></div>
+        </div>
+
+        <div className="container-plattforms">
+          <div classNme="column-one-sixed"></div>
+          <div className="column-four-sixed-posts">
+            {posts.map(post => (
+              <IGPost key={post.shortcode} post={post} />
```
Comment (Urhengulas) [inline frontend/src/components/Result.js:117]: Why are you passing both the `post` and an attribute of it? You kinda will hand over duplicate data.

File: frontend/src/components/Result.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+import {
+  UserSearchServiceClient,
+  UserIdRequest
+} from "../protofiles/usersearch_grpc_web_pb";
+
+export class Result extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: this.props.location.state && this.props.location.state.user,
```
Comment (Urhengulas) [inline frontend/src/components/Result.js:19]: What is that doing?

File: frontend/src/components/Result.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      posts: []
+    };
+  }
+
+  componentDidMount() {
+    const user = this.state.user;
+    if (!user) {
+      console.log(":(");
```
Comment (Urhengulas) [inline frontend/src/components/Result.js]: Please change the msg to sth more descriptive

File: frontend/src/components/Result.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
+import React, { Component } from "react";
+import App, { users, userdata, user } from "../App";
+import { withRouter } from "react-router";
+import IGlogo from "./img/instagram.png";
+import FBlogo from "./img/fb.png";
+import Twitterlogo from "./img/twitter.png";
+import LIlogo from "./img/linkedin.png";
+import IGPost from "./IGPost";
+
```
Comment (Urhengulas) [inline frontend/src/components/Result.js]:
```suggestion
```

File: frontend/src/components/Result.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    }
+    const client = new UserSearchServiceClient("http://localhost:4000");
+    const userIdRequest = new UserIdRequest();
+    userIdRequest.setUserId(user.id);
+    client.getInstaPostsWithUserId(userIdRequest, {}, (err, response) => {
+      if (err) {
+        console.log(err);
+        return;
+      }
+      console.log(response);
```
Comment (Urhengulas) [inline frontend/src/components/Result.js]: More descriptive please

File: frontend/src/components/Result.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+      const posts = response.getInstaPostsList();
+      const postdata = posts
+        .map(posts => ({
+          id: posts.getPostId(),
+          img: posts.getImgUrl(),
+          caption: posts.getCaption(),
+          shortcode: posts.getShortCode()
+        }))
+        .filter(post => !!post.img);
```
Comment (Urhengulas) [inline frontend/src/components/Result.js:46]: isn't the `!!` a double negation and thereby doing nothing in the end?

File: frontend/src/components/Result.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+      // this.props.history.push({
```
Comment (Urhengulas) [inline frontend/src/components/Result.js]: Can you drop that?

Comment (Urhengulas) [reply frontend/src/components/Result.js]: I mean the commented part

File: frontend/src/components/Result.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        <div className="container-plattforms">
+          <div classNme="column-one-sixed"></div>
+          <div className="column-four-sixed-posts">
+            {posts.map(post => (
+              <IGPost key={post.shortcode} post={post} />
+            ))}
+          </div>
+          <div classNme="column-one-sixed"></div>
+        </div>
+      </div>
```
Comment (Urhengulas) [inline frontend/src/components/Result.js:122]: That are a lot of `div`-elements. Is there some way to avoid that? Or to format it more nicely?

**Review by Urhengulas — CHANGES_REQUESTED** [review]

File: frontend/src/components/IGPost.js (hunk 1 above)
Comment (lukas-menzel) [reply frontend/src/components/IGPost.js:4]: destructuring

File: frontend/src/components/Result.js (hunk 2 above)
Comment (lukas-menzel) [reply frontend/src/components/Result.js:117]: .map needs key

File: frontend/src/components/Result.js (hunk 4 above)
Comment (lukas-menzel) [reply frontend/src/components/Result.js]: changed

File: frontend/src/components/Result.js (hunk 7 above)
Comment (lukas-menzel) [reply frontend/src/components/Result.js:46]: results in the boolean value from post.img

File: frontend/src/components/Result.js (hunk 9 above)
Comment (lukas-menzel) [reply frontend/src/components/Result.js:122]: we could use grid components or bootstrap implement rows and columns as own components.

File: frontend/src/components/Result.js (hunk 6 above)
Comment (lukas-menzel) [reply frontend/src/components/Result.js]: Deleted the `console.log`, but a short description can be used to describe the log.

**Review by Urhengulas — APPROVED** [review]

**Pull request merged**